### DISCUSSION TITLE
[Feature]:  Multi-Select Photos

### DISCUSSION THREAD

**Discussion by PixelJonas** [immich-app/immich#1659] [Feature Request]
### Feature detail

As a user I want to be able to quickly select multiple photos to add them to an album
So that I can organize and share my photos quickly with others

When opening the "Add photos" menu on the mobile-app I'd like to Drag-to-Select multiple consecutive images to quickly add them to an album

When selecting Images on the web, after selecting an initial Image, I want to be able to "SHIFT+Click" to select every consecutive photo in between the first selected photo and the one I clicked on. This could also be reversed (I select the last picture and "SHIFT+Click" on the first one"

With multiple photos already selected, this should also work for de-selecting them ("drag-to-select", or "shift-click")

### Platform

Mobile App

**Comment by PaddiM8** [comment]
One of my main problems with Google Photos was how inflexible it was with selecting photos. I want to be able to select them just like I would in a file manager. It would be really nice if Immich would do this better than Google Photos! Organisation is a huge part of backups

**Comment by jagjordi** [comment]
I think this is a major feature that will really make it much easier to create albums. Now its very painful if the number of pictures is big and they are mixed with other stuff in the same date

**Comment by martyfuhry** [comment]
This is a super cool idea!

> When opening the "Add photos" menu on the mobile-app I'd like to Drag-to-Select multiple consecutive images to quickly add them to an album

I can't think how to disambiguate between "Drag-to-Select" and "Drag-to-Scroll" on mobile. Do you have any ideas?

**Reply by arnelap** [reply]
Press and hold while dragging. another very popular photos service does it like that

**Comment by jrasm91** [comment]
The functionally between drag to select and scroll is basically a long press and then scroll (without a release between the two) vs long press to select followed by a release and then independent scroll.

Can you set a flag on long press and clear it on long press release? If the flag is set and scroll is happening we should be selecting.

**Comment by DCCInterstellar** [comment]
Has this feature been implemented into immich or is it still in the process of making? I was able to select pictures but not everything.

**Comment by bcsteeve** [comment]
I noticed I couldn't Shift+click in Immich.  I checked, you can do that in Google Photos.

(oddly, you can't Ctrl+click in GP though.  For the uninitiated, shift+click is for consecutive items while ctrl+click is for non-consecutive)

**Comment by jagjordi** [comment]
I think its probably being worked out, recently they have added the "Select All" button

**Reply by bcsteeve** [reply]
Gotcha.  I'm new here so not sure what's recent :)

**Comment by jagjordi** [comment]
This feature is now on the latest release @bcsteeve

**Reply by bcsteeve** [reply]
Great!  I think it caused a bug though.  I just reported.  https://github.com/immich-app/immich/issues/3115

**Reply by bcsteeve** [reply]
With the bug fixed, the multi-select feature works GREAT!  Nicely done.

**Comment by uhthomas** [comment]
Fixed by #3086.

**Reply by kamzil** [reply]
This is for the web app, not mobile.

**Reply by arnelap** [reply]
Is tis feature removed again? It's not working for me in the webapp 

Edit: NVM, found out you need to hold shift while dragging the mouse. 
I expected it to work as in any other file manager where holding shift while selecting an item selects everything between the earlier selection and that item.

**Reply by DCCInterstellar** [reply]
> Is tis feature removed again? It's not working for me in the webapp 
> 
> 
> 
> Edit: NVM, found out you need to hold shift while dragging the mouse. 
> 
> I expected it to work as in any other file manager where holding shift while selecting an item selects everything between the earlier selection and that item.

There is another way. When you select on a single picture or video. At the top right you will see this little square. Click on that and it'll select all your photos and videos!

Here is what it looks like:
[image](https://github.com/immich-app/immich/assets/124222270/cb3317f8-0a47-49fd-be1c-07c696abb720)

**Reply by kamzil** [reply]
While very useful and a great feature, select all button is only available in the web version, not mobile app.

**Comment by Monarsoft** [comment]
Agree, there can be a one-click option to select all, select info all.

**Comment by kamzil** [comment]
Is this supposed to be implemented in the Android app? I can't find it. I can only select a group at a time in timeline view, and one photo at a time in album view. Neither select all button nor long-press and drag to quickly select multiple photos.

**Reply by kamzil** [reply]
Therefore, could this be reopened please?

**Comment by arnelap** [comment]
This doesn't work in search results?? That would be the main place where I use it, as I'm searching for something and then deleting or archiving or moving to album a selection of it.

**Comment by arnelap** [comment]
Does the current shit drag only work on the home screen? 
It seems to not work in Search results. 

Also, tbh I expected it to work like in any file manager where it's shift+click instead of shift+drag. 

Can this be re-opened to make it work everywhere, not just the home screen?
And of course on mobile as well (hold + drag)

**Reply by Tmatz27** [reply]
I agree that it is not working in the search screen and that is where i would primarily want to use it

**Comment by bo0tzz** [comment]
This is now supported in the mobile app, and in the timeline view on the web. Adding multi-select to the other views on the web will be tracked in #9159

**Reply by kamzil** [reply]
Awesome! Long-press multi-select is working now in the Android app. Thanks for making it happen.

However, I still can't find the "select all" button in the mobile app. Is that a separate feature with its own discussion? Currently it's nearly impossible to select all the photos in timeline view, as it's not practical to long-press-drag over thousands of photos.

**Reply by rsutton1** [reply]
@kamzil I'm looking for a "select all" button in the Android app as well, I can't find it. I don't think it's implemented yet.

**Comment by ocean-haiyang** [comment]
We really need this for basic immich in browser!
